src/components/Jobs/JobDetail.tsx

```diff
@@ -1,4 +1,73 @@
-function JobDetail() {
-  return <div className="flex h-full flex-col" />;
+import { Button, Container, Loader, Tabs } from '@mantine/core';
+import { useParams } from 'react-router-dom';
+
+import JobDescriptionTab from '@/components/Jobs/JobDescriptionTab';
+import JobOverviewTab from '@/components/Jobs/JobOverviewTab';
+import { useAuth } from '@/hooks/useAuth';
+import { useJob, useKillJobMutation } from '@/hooks/useJobs';
+
+export default function JobDetails() {
+  const { jobId = '', region = '', env = '' } = useParams();
+  const { data: jobData } = useJob(env, region, jobId);
+  const { mutate } = useKillJobMutation();
+  const { user } = useAuth();
+
+  const tabs: string[] = ['Overview', 'Job Description', 'Stage Details', 'Job Status Output'];
+  const items = tabs.map((tab) => (
+    <Tabs.Tab className={`rounded-lg bg-transparent text-gray-500`} value={tab} key={tab}>
+      {tab}
+    </Tabs.Tab>
+  ));
+
+  return (
+    <div className={'w-full'}>
+      {jobData === undefined && (
```

Comment:
Instead of checking jobData for non nulll every time time why not do this...

if (!jobData) {
   return <div className={'w-full'}>
                  <div className={'flex h-screen w-full flex-row items-center justify-center'}>
                      <Loader />
                    </div>
              </div>
} else {
  <div className={'w-full'}>
      // Rest of logic where we know jobData is non-null
  </div>


}